src/main/java/com/mathworks/ci/MatlabReleaseInfo.java

```diff
@@ -107,14 +106,31 @@ public boolean verLessThan(double version) throws MatlabVersionNotFoundException
                 else if(!this.matlabRoot.exists()){
                     throw new NotDirectoryException("Invalid matlabroot path");
                 }else {
-                    versionInfoCache.putAll(VERSION_OLDER_THAN_17A);
-                }
-                
+					// Get the version information from Contents.m file when VersionInfo.xml is not
+					// present.
+					FilePath contentFile = new FilePath(this.matlabRoot, CONTENTS_FILE);
+					String actualVersion = null;
+					try (InputStream in = contentFile.read();
+							BufferedReader br = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8))) {
```

Comment:
As mentioned earlier, MATLAB files will be UTF-8 encoded in 20a or later. In older releases MATLAB m files are encoded in the "default encoding". However MATLAB's notion of "default encoding" may not always match the default encoding that the Java code in the Jenkins plugin will use, and it will be hard to get this information without starting MATLAB.

However, if Contents.m files contain ASCII only content, then we don't have to worry about the encoding here. Hence UTF-8 is a safe choice.